src/elements/play-assets/file-upload-dropper.ts

```diff
@@ -0,0 +1,268 @@
+import {
+  css,
+  html,
+  LitElement,
+  type PropertyValues,
+  type TemplateResult
+} from 'lit'
+import {customElement, property, state} from 'lit/decorators.js'
+import {classMap} from 'lit-html/directives/class-map.js'
+import {ifDefined} from 'lit-html/directives/if-defined.js'
+import {
+  fileAccessContext,
+  type FilePickerType,
+  flattenAcceptTypes,
+  hasFileAccessAPI,
+  tryGetAsFilesystemHandle
+} from '../../utils/file-access-api.js'
+import {cssReset} from '../../utils/css-reset.js'
+import {Bubble} from '../../utils/bubble.js'
+import {PlayAssets} from './play-assets.js'
+
+declare global {
+  interface HTMLElementEventMap {
+    'files-selected': CustomEvent<FilesSelectedEvent>
+    cancelled: CustomEvent<never>
+  }
+  interface HTMLElementTagNameMap {
+    'file-upload-dropper': FileUploadDropper
+  }
+}
+
+export type FilesSelectedEvent = {
```

Comment:
consider calling this just `FileSelection` or something like that. you don't get an Event until you wrap it in `CustomEvent<FileSelection>`.